test/forms/PropertyOrLandDescriptionFormProviderSpec.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import forms.behaviours.StringFieldBehaviours
+import forms.property_or_land.PropertyOrLandDescriptionFormProvider
+import play.api.data.FormError
+
+class PropertyOrLandDescriptionFormProviderSpec extends StringFieldBehaviours {
+
+  val requiredKey = "propertyOrLandDescription.error.required"
+  val lengthKey = "propertyOrLandDescription.error.length"
+  val maxLength = 56
+
+  val form = new PropertyOrLandDescriptionFormProvider()()
+
+  ".value" must {
+
+    val fieldName = "value"
+
+    behave like fieldThatBindsValidData(
+      form,
+      fieldName,
+      stringsWithMaxLength(maxLength)
+    )
+
+    behave like fieldWithMaxLength(
+      form,
+      fieldName,
+      maxLength = maxLength,
+      lengthError = FormError(fieldName, lengthKey, Seq(maxLength))
+    )
+
+    behave like mandatoryField(
+      form,
+      fieldName,
+      requiredError = FormError(fieldName, requiredKey)
+    )
```

Comment:
Are we happy these behaviours give the correct coverage? Do we want to use a Regex generator?

Reply:
Added further tests with regex generator